Edit channel app toolbars layout + staging tree toolbar

MisRob opened the pull request:
## Description

The first commit contains a minimal set of updates necessary to allow me to add a toolbar to the staging channel page. The solution is explained in more detail in issue #1985. The second commit then adds the toolbar.

#### Issue Addressed (if applicable)

Closes #1985

#### Before/After Screenshots (if applicable)

![toolbar](https://user-images.githubusercontent.com/13509191/88335016-ee66e100-cd32-11ea-8ce0-53c0419e5813.png)

## Steps to Test

- [ ] Login as admin (username: a@a.com, password: a) to demo server
- [ ] Visit http://misrob-app-layout-toolbar.studio.cd.learningequality.org/channels/a9cf05d53fb15eac937eb3a1020856fa/#/staging/d3919da316fa42718990604eda45397e

## Checklist

- [x] Is the code clean and well-commented?
- [ ] Has the `docs` label been added if this introduces a change that needs to be updated in the [user docs](https://kolibri-studio.readthedocs.io/en/latest/index.html)?
- [ ] Has the `CHANGELOG` label been added to this pull request? Items with this label will be added to the [CHANGELOG](https://github.com/learningequality/studio/blob/master/CHANGELOG.md) at a later time
- [x] Are there tests for this change?
- [ ] Are all user-facing strings translated properly (if applicable)?
- [ ] Has the `notranslate` class been added to elements that shouldn't be translated by Google Chrome's automatic translation feature (e.g. icons, user-generated text)?
- [ ] Are all UI components LTR and RTL compliant (if applicable)?
- [ ] Are views organized into `pages`, `components`, and `layouts` directories [as described in the docs](https://github.com/learningequality/studio/blob/vue-refactor/docs/architecture.md#where-does-the-frontend-code-live)?
- [ ] Are there any new ways this uses user data that needs to be factored into our [Privacy Policy](https://github.com/learningequality/studio/tree/master/contentcuration/contentcuration/templates/policies/text)?
- [ ] Are there any new interactions that need to be added to the [QA Sheet](https://docs.google.com/spreadsheets/d/1HF4Gy6rb_BLbZoNkZEWZonKFBqPyVEiQq4Ve6XgIYmQ/edit#gid=0)?
- [ ] Are there opportunities for using Google Analytics here (if applicable)?
- [ ] If the Pipfile has been changed, is the updated Pipfile.lock file also included in this PR?
- [ ] Are the migrations [safe for a large db](https://www.braintreepayments.com/blog/safe-operations-for-high-volume-postgresql/) (if applicable)?

jayoshih commented on contentcuration/contentcuration/frontend/channelEdit/views/TreeView/TreeViewBase.vue:
```diff
@@ -0,0 +1,228 @@
+              <VListTile v-if="canEdit" :to="editChannelLink">
+                <VListTileTitle>{{ $tr('editChannel') }}</VListTileTitle>
+              </VListTile>
+              <VListTile v-if="canEdit" @click.stop>
```
This should navigate to the trash moddal

jayoshih commented on contentcuration/contentcuration/frontend/channelEdit/views/TreeView/index.vue:
```diff
@@ -1,78 +1,80 @@
+          <CurrentTopicView :topicId="nodeId" :detailNodeId="detailNodeId" />
+        </VContainer>
+        <router-view />
+        <ImportContentProgressModal
```
Can remove this now

jayoshih commented on contentcuration/contentcuration/frontend/channelEdit/pages/StagingTreeView/index.vue:
```diff
@@ -1,229 +1,248 @@
+            </VLayout>
+          </VContainer>
+
+          <BottomBar>
```
May need to use app on the toolbar or add padding to the Container element as the last item in the list gets partially blocked by the toolbar
![image](https://user-images.githubusercontent.com/7447496/88349283-74b7fe80-cd04-11ea-8c79-4650ffa09d5b.png)

jayoshih commented on contentcuration/contentcuration/frontend/channelEdit/pages/StagingTreeView/index.vue:
```diff
@@ -1,229 +1,248 @@
+                    <VCardText>
+                      <DiffTable :stagingDiff="stagingDiff" />
+                    </VCardText>
+                    <VDivider />
```
Might need to double check with design on this, but other modals don't have a divider above the actions. Might want to remove for consistency
![image](https://user-images.githubusercontent.com/7447496/88349358-acbf4180-cd04-11ea-9ab4-fb415a51b1a1.png)
![image](https://user-images.githubusercontent.com/7447496/88349391-c6f91f80-cd04-11ea-83b7-2d57d5bfe5ad.png)

jayoshih commented on contentcuration/contentcuration/frontend/channelEdit/pages/StagingTreeView/index.vue:
```diff
@@ -1,229 +1,248 @@
-                  v-on="on"
+              </VList>
+
+              <ResourceDrawer
```
Should the resource drawer extend over the breadcrumbs as well? It seems off to have the drawer border not connect with the toolbar border
![image](https://user-images.githubusercontent.com/7447496/88349466-045dad00-cd05-11ea-93bf-75c0a11c5e10.png)

jayoshih commented on contentcuration/contentcuration/frontend/channelEdit/pages/StagingTreeView/index.vue:
```diff
@@ -1,229 +1,248 @@
+    <MainNavigationDrawer v-model="drawer" />
+
+    <VContent>
+      <VContainer fluid fill-height class="pa-0">
```
I need to double check on this, but I think the topic tree hierarchy drawer, resource drawer, and current topic list should all have their own scrollbars instead of one for the whole page. Probably not a big deal if we follow this up in a separate PR though
![image](https://user-images.githubusercontent.com/7447496/88349580-54d50a80-cd05-11ea-8b40-b0bd7cb1371c.png)

MisRob replied:
That makes sense. I opened [an issue]([url](#2000
)) because I guess we should update the studio tree component and the resource drawer component that will also affect other places in the app.

MisRob commented:
Thank you, @jayoshih. I've resolved all comments and opened an issue for scrollbars. When checking missing breadcrumbs border and blocked the last item in the content nodes list, I noticed more layout problems, so I added a commit to update the whole page layout that also addresses those issues.

jayoshih commented on contentcuration/contentcuration/frontend/channelEdit/views/TreeView/index.vue:
```diff
@@ -1,78 +1,75 @@
+              :style="{'text-decoration': 'underline'}"
+              data-test="staging-tree-link"
+            >{{ $tr('updatedResourcesReadyForReview') }}</router-link>
+            (<time :datetime="channelModifiedDate">{{ prettyChannelModifiedDate }}</time>)
```
Will need to figure out a different way of handling this to support i18n

MisRob replied:
@jayoshih I am not sure what's meant by this comment exactly, we're using `vue-intl`, isn't that sufficient?
